components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.v1/src/main/java/org/wso2/carbon/identity/api/server/application/management/v1/core/functions/application/inbound/oauth2/OAuthInboundFunctions.java

```diff
@@ -223,20 +223,17 @@ public static void revokeOAuthClient(String clientId) {
      *
      * @param applicationId   application id
      * @param oidcConfigModel oidc configurations of the application.
+     * @throws CORSManagementServiceException if CORS origin update fails.
      */
-    public static void updateCorsOrigins(String applicationId, OpenIDConnectConfiguration oidcConfigModel) {
+    public static void updateCorsOrigins(String applicationId, OpenIDConnectConfiguration oidcConfigModel)
+            throws CORSManagementServiceException {
 
         String tenantDomain = ContextLoader.getTenantDomainFromContext();
-
         // Update the CORS origins.
         List<String> corsOrigins = oidcConfigModel.getAllowedOrigins();
-        try {
-            if (corsOrigins != null) {
-                ApplicationManagementServiceHolder.getCorsManagementService()
-                        .setCORSOrigins(applicationId, corsOrigins, tenantDomain);
-            }
-        } catch (CORSManagementServiceException e) {
-            throw handleException(e);
+        if (corsOrigins != null) {
```

Comment:
Better to use CollectionUtils.isEmpty(list); instead of null check. from import org.apache.commons.collections.CollectionUtils;. Would depend on the behavior of who the result is returned by oidcConfigModel.